app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -83,6 +89,14 @@ public function setTriggerModel(TriggerModel $triggerModel)
     }
 
     /**
+     * @param EventDispatcherInterface $dispatcher
+     */
+    public function setDispatcher(EventDispatcherInterface $dispatcher)
```

Comment:
I do not like setter injection. It is very easy to forgot to call this and the code will not work as expected.
This has to be injected in a constructor which means that we need to define this class as a service (which is not so easy in current Mautic configuration) or move the code to Model class (which is best solution I think - repository should have only one responsibility like other classes.